core/src/main/scala/io/qbeast/spark/index/OTreeDataAnalyzer.scala

```diff
@@ -216,6 +217,21 @@ object DoublePassOTreeDataAnalyzer
       indexStatus: IndexStatus,
       options: QbeastOptions): (DataFrame, TableChanges) = {
     logTrace(s"Begin: Analyzing the input data with existing revision: ${indexStatus.revision}")
+
+    // Check if the DataFrame is deterministic
+    logDebug(s"Checking the determinism of the input data")
+    val isSourceDeterministic =
+      analyzeDataFrameDeterminism(dataFrame, indexStatus.revision)
```

Comment:
Use `revisionToUse` instead of the existing revision in case we want to allow changing indexing columns

Reply:
This part is a bit challenging. While I could use `revisionToUse`, it would defeat the purpose of avoiding the computation of min/max when we already know the source is non-deterministic.

Alternatively, it would be ideal to allow users to provide their own `columnStats` and enforce their use instead of relying on DataFrame analysis. This approach would give users more control and help bypass unnecessary computations when the source characteristics are already known.

What is your take with the current code status?